lib/Service/ValueObject/ViewColumnInformation.php

```diff
@@ -17,28 +17,40 @@
 class ViewColumnInformation implements ArrayAccess, JsonSerializable {
 	public const KEY_ID = 'columnId';
 	public const KEY_ORDER = 'order';
+	public const KEY_READONLY = 'readonly';
 
-	/** @var array{columndId?: int, order?: int} */
+	/** @var array{columndId?: int, order?: int, readonly?: bool} */
 	protected array $data = [];
 	protected const KEYS = [
 		self::KEY_ID,
 		self::KEY_ORDER,
+		self::KEY_READONLY,
 	];
 
 	public function __construct(
 		int $columnId,
 		int $order,
+		bool $readonly = false,
 	) {
 		$this->offsetSet(self::KEY_ID, $columnId);
 		$this->offsetSet(self::KEY_ORDER, $order);
+		$this->offsetSet(self::KEY_READONLY, $readonly);
+	}
+
+	public function getId(): int {
+		return $this->offsetGet(self::KEY_ID);
 	}
 
 	public function getOrder(): int {
 		return $this->offsetGet(self::KEY_ORDER);
 	}
 
+	public function isReadonly(): bool {
+		return $this->offsetGet(self::KEY_READONLY) ?? false;
+	}
+
 	public static function fromArray(array $data): static {
-		$vci = new static($data[self::KEY_ID], $data[self::KEY_ORDER]);
+		$vci = new static($data[self::KEY_ID], $data[self::KEY_ORDER], $data[self::KEY_READONLY] ?? false);
```

Comment:
If we add all parameters into the constructor, please split this to multiple lines  to gain easier diffs in futures.

Reply:
done